.drone.yml

```diff
@@ -0,0 +1,199 @@
+workspace:
+  base: /var/www/owncloud
+  path: apps/twofactor_totp
+
+branches:
+  - master
+
+pipeline:
+  install-server:
+    image: owncloudci/core
+    pull: true
+    version: ${OC_VERSION}
+    db_type: ${DB_TYPE}
+    db_name: ${DB_NAME=owncloud}
+    db_host: ${DB_TYPE}
+    db_username: autotest
+    db_password: owncloud
+    when:
+      matrix:
+        NEED_SERVER: true
+
+  owncloud-log:
+    image: owncloud/ubuntu:16.04
+    detach: true
+    pull: true
+    commands:
+      - tail -f /var/www/owncloud/data/owncloud.log
+    when:
+      matrix:
+        NEED_SERVER: true
+
+  install-app:
+    image: owncloudci/php:${PHP_VERSION}
+    pull: true
+    commands:
+      - cd /var/www/owncloud/apps/twofactor_totp
+      - make
+      - cd /var/www/owncloud/
+      - php occ a:l
+      - php occ a:e twofactor_totp
+      - php occ a:e testing
+      - php occ a:l
+      - php occ config:system:set trusted_domains 1 --value=owncloud
+      - php occ log:manage --level 0
+    when:
+      matrix:
+        NEED_INSTALL_APP: true
+
+  fix-permissions:
+    image: owncloudci/php:${PHP_VERSION}
+    pull: true
+    commands:
+      - chown www-data /var/www/owncloud -R
+    when:
+      matrix:
+        NEED_INSTALL_APP: true
+
+  owncloud-coding-standard:
+    image: owncloudci/php:${PHP_VERSION}
+    pull: true
+    commands:
+      - make test-php-style
+    when:
+      matrix:
+        TEST_SUITE: owncloud-coding-standard
+
+  phpunit-tests:
+    image: owncloudci/php:${PHP_VERSION}
+    pull: true
+    environment:
+    - PHP_VERSION=${PHP_VERSION}
+    - COVERAGE=${COVERAGE}
+    commands:
+    - if [ -z "${COVERAGE}" ]; then make test-php-unit; fi
+    - if [ "${COVERAGE}" = "true" ]; then make test-php-unit-dbg; fi
+    when:
+      matrix:
+        TEST_SUITE: phpunit
+
+services:
+  mysql:
+    image: mysql:5.5
+    environment:
+    - MYSQL_USER=autotest
+    - MYSQL_PASSWORD=owncloud
+    - MYSQL_DATABASE=${DB_NAME=owncloud}
+    - MYSQL_ROOT_PASSWORD=owncloud
+    when:
+      matrix:
+        DB_TYPE: mysql
+
+  pgsql:
+    image: postgres:9.4
+    environment:
+    - POSTGRES_USER=autotest
+    - POSTGRES_PASSWORD=owncloud
+    - POSTGRES_DB=${DB_NAME=owncloud}
+    when:
+      matrix:
+        DB_TYPE: pgsql
+
+  oci:
+    image: deepdiver/docker-oracle-xe-11g
+    environment:
+    - ORACLE_USER=system
+    - ORACLE_PASSWORD=oracle
+    - ORACLE_DB=${DB_NAME=owncloud}
+    when:
+      matrix:
+        DB_TYPE: oci
+
+  owncloud:
+    image: owncloudci/php:${PHP_VERSION}
+    pull: true
+    environment:
+      - APACHE_WEBROOT=/var/www/owncloud/
+    command: ["/usr/local/bin/apachectl", "-e", "debug", "-D", "FOREGROUND"]
+    when:
+      matrix:
+        NEED_SERVER: true
+
+matrix:
+  include:
+  # owncloud-coding-standard
+  - PHP_VERSION: 7.2
+    TEST_SUITE: owncloud-coding-standard
+
+  # Unit Tests master
+  - PHP_VERSION: 7.1
+    OC_VERSION: daily-master-qa
+    TEST_SUITE: phpunit
+    DB_TYPE: sqlite
```

Comment:
I think sqlite will need an entry in the services section?
We should ask if we should actually bother running with sqlite any more.